jbpm-workitems/jbpm-workitems-webservice/src/main/java/org/jbpm/process/workitem/webservice/WebServiceWorkItemHandler.java

```diff
@@ -533,11 +540,94 @@ public void abortWorkItem(WorkItem workItem,
+        }
+        return cl;
+    }
+
+    private static class CXFJavaCompileClassLoader extends URLClassLoader {
+
+        private URL[] jarUrls;
+
+        public CXFJavaCompileClassLoader(Collection<File> files, ClassLoader parent) {
+            super(new URL[0], parent);
+            // maybe it makes sense to filter only jaxb ones (there are 256 jars in kie-server)?
+            this.jarUrls = files.stream().map(CXFJavaCompileClassLoader::toUrl).toArray(URL[]::new);
+        }
+        
+        @Override
+        public URL[] getURLs() {
+            return jarUrls;
+        }
+
+        private static URL toUrl(File file) {
+            try {
+                return file.toURI().toURL();
+            } catch (MalformedURLException e) {
+                throw new IllegalStateException("Problem converting file to URL: "+file,e);
+            }
         }
+    }
+
+    private void getJarsFromModuleClassLoader(ClassLoader cl, Collection<File> collector) throws ReflectiveOperationException {
+        /* This method makes assumptions over the internal structure of ModuleClassLoader. If this class is changed, this method
+         * will need to change accordingly
+         */
+        AtomicReference paths = (AtomicReference) get(cl, "paths");
+        Object sourceList = get(paths.get(), "sourceList");
+        int size = Array.getLength(sourceList);
+        Method getVFSResource = null;
+        Method getPhysicalFile = null;
+        Field rootField = null;
+        Field rootNameField = null;
+        for (int i = 0; i < size; i++) {
+            Object resource = Array.get(sourceList, i);
+            if (getVFSResource == null) {
+                getVFSResource = resource.getClass().getDeclaredMethod("getResourceLoader");
+                getVFSResource.setAccessible(true);
+            }
+            resource = getVFSResource.invoke(resource);
+            if (rootField == null) {
+                Class<?> resourceClass = resource.getClass();
+                rootField = resourceClass.getDeclaredField("root");
+                rootNameField = resourceClass.getDeclaredField("rootName");
+                rootField.setAccessible(true);
+                rootNameField.setAccessible(true);
+            }
+            String rootName = (String) rootNameField.get(resource);
+            if (rootName.endsWith("jar")) {
+                Object root = rootField.get(resource);
+                if (getPhysicalFile == null) {
+                    getPhysicalFile = root.getClass().getDeclaredMethod("getPhysicalFile");
+                    getPhysicalFile.setAccessible(true);
+                }
+                collector.add(new File(((File) getPhysicalFile.invoke(root)).getParentFile(), rootName));
+            }
+        }
+    }
 
-        return Thread.currentThread().getContextClassLoader();
+    private Object get(Object container, String fieldName) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Maybe a more descriptive method name can be useful? something like `getFieldObject`?

Reply:
sure, will change